cirq/ops/pauli.py

```diff
@@ -11,29 +11,31 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import abc
+from typing import Any, Union, overload, TYPE_CHECKING
 
-from typing import Optional, Union, overload, TYPE_CHECKING
-
-import numpy as np
-
-from cirq import protocols, value
-from cirq.ops import common_gates
-from cirq.type_workarounds import NotImplementedType
+from cirq import value
+from cirq.ops import common_gates, eigen_gate
 
 if TYPE_CHECKING:
     # pylint: disable=unused-import
     from typing import Tuple
 
 
-@value.value_equality
-class Pauli:
-    """Represents the X, Y, or Z axis of the Bloch sphere."""
+class Pauli(eigen_gate.EigenGate, metaclass=abc.ABCMeta):
+    """Represents the Pauli gates.
+
+    This is an abstract class with no public subclasses. The only instances
+    of private subclasses are the X, Y, or Z Pauli gates defined below.
+    """
     X = None  # type: Pauli
```

Comment:
Delete the X/Y/Z attributes. Just `cirq.X`, no `cirq.Pauli.X`.

Not sure if the combined `XYZ` should stay or go.

Reply:
Renaming it to pauli gates is fine.

Reply:
Done.